Error while checking build/O2/fullCI for beaf2a39bb93503e53dadff4a12e0de99472f2c4 at 2022-09-16 01:23:
```
## sw/BUILD/o2checkcode-latest/log
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/Common/test/testGPUsortCUDA.cu:22:10: error: 'boost/test/unit_test.hpp' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUTracking/TRDTracking/GPUTRDTracker.cxx:37:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUTracking/Base/GPUReconstruction.cxx:37:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUTracking/Base/GPUReconstructionCPU.cxx:45:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUTracking/display/GPUDisplay.cxx:36:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUTracking/Base/cuda/GPUReconstructionCUDAGenRTC.cu:16:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUbenchmark/cuda/../Shared/Utils.h:26:10: error: 'boost/program_options.hpp' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUbenchmark/cuda/../Shared/Utils.h:26:10: error: 'boost/program_options.hpp' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/Framework/Logger/include/Framework/Logger.h:14:10: error: 'fairlogger/Logger.h' file not found [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:520:12: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:1059:69: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:1175:5: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:1852:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:2276:18: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:4336:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/sm_20_atomic_functions.h:89:39: error: redefinition of 'atomicAdd' [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/Framework/Logger/include/Framework/Logger.h:14:10: error: 'fairlogger/Logger.h' file not found [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:520:12: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:1059:69: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:1175:5: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:1852:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:2276:18: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:4336:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/sm_20_atomic_functions.h:89:39: error: redefinition of 'atomicAdd' [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUbenchmark/hip/../Shared/Utils.h:146:34: error: use of undeclared identifier 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:8: error: unknown type name '__host__' [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:27: error: expected ';' after top level declarator [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:33: error: overloaded 'operator+=' must have at least one parameter of class or enumeration type [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:44: error: unknown type name 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:53: error: unknown type name 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:66:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:66:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:67:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:67:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:68:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:68:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:69:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:69:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:85:1: error: unknown type name '__global__' [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:19: error: use of undeclared identifier 'blockIdx' [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:32: error: use of undeclared identifier 'blockDim' [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:45: error: use of undeclared identifier 'threadIdx' [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:78: error: use of undeclared identifier 'blockDim' [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUbenchmark/hip/../Shared/Utils.h:146:34: error: use of undeclared identifier 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/GPU/GPUbenchmark/hip/benchmark.hip.cxx:199:35: error: use of undeclared identifier 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/Detectors/EMCAL/calibration/include/EMCALCalibration/EMCALCalibExtractor.h:34:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/Detectors/EMCAL/calibration/include/EMCALCalibration/EMCALCalibExtractor.h:34:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/Detectors/EMCAL/calibration/include/EMCALCalibration/EMCALCalibExtractor.h:34:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/9841-slc8_x86-64/0/Detectors/TOF/calibration/src/TOFChannelCalibrator.cxx:23:10: error: 'omp.h' file not found [clang-diagnostic-error]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9841/beaf2a39bb93503e53dadff4a12e0de99472f2c4/build_O2_fullCI/pretty.html).

--- a/DataFormats/Detectors/MUON/MID/include/DataFormatsMID/Track.h
+++ b/DataFormats/Detectors/MUON/MID/include/DataFormatsMID/Track.h
@@ -143,19 +143,26 @@ class Track
   /// Gets the word allowing to compute the chamber efficiency
   uint32_t getEfficiencyWord() const { return mEfficiencyWord; }
 
-  /// Sets the fired local board for efficiency calculation
-  /// \param locId local board ID in the range 1-234
-  void setFiredLocalBoard(int locId) { setEfficiencyWord(8, 0xFF, locId); }
+  /// Sets the fired FEE ID
+  void setFiredFEEId(int uniqueFeeId) { setEfficiencyWord(8, 0x7FFF, uniqueFeeId); }
+
+  /// Gets the fired FEE ID
+  int getFiredFEEId() const { return (mEfficiencyWord >> 8) & 0x7FFF; }
 
   /// Gets the fired local board for efficiency calculation
-  int getFiredLocalBoard() const { return (mEfficiencyWord >> 8) & 0xFF; }
+  [[deprecated]] int getFiredLocalBoard() const { return (mEfficiencyWord >> 8) & 0xFF; }
 
-  /// Sets the fired detection element ID for efficiency calculation
-  /// \param deId detection element ID in the range 1-234
-  void setFiredDeId(int deId) { setEfficiencyWord(16, 0xFF, deId); }
+  /// Gets the fired Detection Element ID
+  int getFiredDEId() const { return (mEfficiencyWord >> 16) & 0x7F; }
 
-  /// Gets the fired detection element ID for efficiency calculation
-  int getFiredDeId() const { return (mEfficiencyWord >> 16) & 0xFF; }
+  /// Gets the fired Detection Element ID
+  [[deprecated("Use getFiredDEId instead")]] int getFiredDeId() const { return getFiredDEId(); }
+
+  /// Gets the fired column ID
+  int getFiredColumnId() const { return (mEfficiencyWord >> 12) & 0x7; }
+
+  /// Gets the fired line ID
+  int getFiredLineId() const { return (mEfficiencyWord >> 8) & 0x3; }
 
   /// Sets the flag for efficiency calculation
   /// \param effFlag efficiency flag

--- a/DataFormats/Detectors/MUON/MID/src/Track.cxx
+++ b/DataFormats/Detectors/MUON/MID/src/Track.cxx
@@ -177,7 +177,7 @@ std::ostream& operator<<(std::ostream& stream, const Track& track)
     stream << ((ival == 5) ? ")" : ", ");
   }
   stream << fmt::format(" chi2/ndf: {:g}/{:d}", track.getChi2(), track.getNDF());
-  stream << fmt::format(" hitMap: 0x{:x} locId: {:d} deId: {:d} effFlag {:d}", track.getHitMap(), track.getFiredLocalBoard(), track.getFiredDeId(), track.getEfficiencyFlag());
+  stream << fmt::format(" hitMap: 0x{:x} deId: {:d} columnId: {:d} lineId: {:d} effFlag {:d}", track.getHitMap(), track.getFiredDEId(), track.getFiredColumnId(), track.getFiredLineId(), track.getEfficiencyFlag());
   return stream;
 }
 

--- a/Detectors/MUON/MID/Base/include/MIDBase/DetectorParameters.h
+++ b/Detectors/MUON/MID/Base/include/MIDBase/DetectorParameters.h
@@ -58,6 +58,29 @@ int getDEId(bool isRight, int chamber, int rpc);
 /// Gets the detection element name from its ID
 /// \param deId The detection element ID
 std::string getDEName(int deId);
+
+/// Gets the unique Front-End Electronics ID
+/// \param deId The detection element ID
+/// \param columnId Column ID
+/// \param lineId Line in column
+/// \return unique FEE ID
+inline int getUniqueFEEId(int deId, int columnId, int lineId) { return lineId | (columnId << 4) | (deId << 8); }
+
+/// Gets the detection element ID from the unique FEE ID
+/// \param uniqueFEEId Unique FEE ID
+/// \return Detection element ID
+inline int getDEIdFromFEEId(int uniqueFEEId) { return (uniqueFEEId >> 8) & 0x7F; }
+
+/// Gets the column ID from the unique FEE ID
+/// \param uniqueFEEId Unique FEE ID
+/// \return Column ID
+inline int getColumnIdFromFEEId(int uniqueFEEId) { return (uniqueFEEId >> 4) & 0x7; }
+
+/// Gets the line ID from the unique FEE ID
+/// \param uniqueFEEId Unique FEE ID
+/// \return Line ID
+inline int getLineIdFromFEEId(int uniqueFEEId) { return uniqueFEEId & 0x3; }
+
 } // namespace detparams
 } // namespace mid
 } // namespace o2

--- a/Detectors/MUON/MID/Tracking/include/MIDTracking/HitMapBuilder.h
+++ b/Detectors/MUON/MID/Tracking/include/MIDTracking/HitMapBuilder.h
@@ -18,9 +18,10 @@
 #define O2_MID_HITMAPBUILDER_H
 
 #include <vector>
-#include <unordered_set>
+#include <unordered_map>
 #include <gsl/gsl>
 #include "DataFormatsMID/Cluster.h"
+#include "DataFormatsMID/ColumnData.h"
 #include "DataFormatsMID/ROFRecord.h"
 #include "DataFormatsMID/Track.h"
 #include "MIDBase/GeometryTransformer.h"
@@ -52,6 +53,10 @@ class HitMapBuilder
   /// \param clusters gsl::span of associated clusters (in local coordinates)
   void process(std::vector<Track>& tracks, gsl::span<const Cluster> clusters) const;
 
+  /// Sets the masked channels
+  /// \param maskedChannels vector of masked channels
+  void setMaskedChannels(const std::vector<ColumnData>& maskedChannels);
+
  private:
   /// Checks if the track crossed the same element
   /// \param fired Vector of fired elements
@@ -74,8 +79,14 @@ class HitMapBuilder
   /// \return The FEE ID in MT11
   int getFEEIdMT11(double xp, double yp, uint8_t deId) const;
 
-  Mapping mMapping;     ///< Mapping
-  HitFinder mHitFinder; ///< Hit finder
+  /// Cluster matches masked channel
+  /// \param cl Cluster
+  /// \return true if the cluster matches the masked channel
+  bool matchesMaskedChannel(const Cluster& cl) const;
+
+  Mapping mMapping;                                             ///< Mapping
+  HitFinder mHitFinder;                                         ///< Hit finder
+  std::unordered_map<int, std::vector<MpArea>> mMaskedChannels; ///< Masked channels
 };
 } // namespace mid
 } // namespace o2

--- a/Detectors/MUON/MID/Tracking/include/MIDTracking/HitMapBuilder.h
+++ b/Detectors/MUON/MID/Tracking/include/MIDTracking/HitMapBuilder.h
@@ -65,14 +65,14 @@ class HitMapBuilder
   /// \param firedLocIds Vector of fired Local board Ids
   /// \param nonFiredLocIds Vector of non-fired Local board Ids
   /// \return the efficiency flag
-  int getEffFlag(const std::vector<int>& firedRPCLines, const std::vector<int>& nonFiredRPCLines, const std::vector<int>& firedLocIds, const std::vector<int>& nonFiredLocIds) const;
+  int getEffFlag(const std::vector<int>& firedFEEIdMT11, const std::vector<int>& nonFiredFEEIdMT11) const;
 
-  /// Function to extract the local board ID of the cluster
-  /// \param xp cluster x coordinate
-  /// \param yp cluster y coordinate
-  /// \param deId cluster Detection Element ID
-  /// \return the Local board ID
-  int getLocId(double xp, double yp, uint8_t deId) const;
+  /// Returns the FEE ID in MT11
+  /// \param xp x position
+  /// \param yp y position
+  /// \param deId Detector element ID
+  /// \return The FEE ID in MT11
+  int getFEEIdMT11(double xp, double yp, uint8_t deId) const;
 
   Mapping mMapping;     ///< Mapping
   HitFinder mHitFinder; ///< Hit finder

--- a/Detectors/MUON/MID/Tracking/src/HitMapBuilder.cxx
+++ b/Detectors/MUON/MID/Tracking/src/HitMapBuilder.cxx
@@ -14,12 +14,33 @@
 /// \author Diego Stocco <Diego.Stocco at cern.ch>
 /// \date   10 December 2021
 #include "MIDTracking/HitMapBuilder.h"
+
+#include <array>
 namespace o2
 {
 namespace mid
 {
 
 HitMapBuilder::HitMapBuilder(const GeometryTransformer& geoTrans) : mMapping(), mHitFinder(geoTrans) {}
+
+void HitMapBuilder::setMaskedChannels(const std::vector<ColumnData>& maskedChannels)
+{
+  mMaskedChannels.clear();
+  std::array<int, 2> nLines{4, 1};
+  for (auto& mask : maskedChannels) {
+    for (int icath = 0; icath < 2; ++icath) {
+      for (int iline = 0; iline < nLines[icath]; ++iline) {
+        auto pat = mask.getPattern(icath, iline);
+        for (int istrip = 0; istrip < detparams::NStripsBP; ++istrip) {
+          if (pat & (1 << istrip)) {
+            auto area = mMapping.stripByLocation(istrip, icath, iline, mask.columnId, mask.deId);
+            mMaskedChannels[mask.deId].emplace_back(area);
+          }
+        }
+      }
+    }
+  }
+}
 
 bool HitMapBuilder::crossCommonElement(const std::vector<int>& fired, const std::vector<int>& nonFired) const
 {
@@ -75,35 +96,55 @@ int HitMapBuilder::getFEEIdMT11(double xp, double yp, uint8_t deId) const
   return -1;
 }
 
+bool HitMapBuilder::matchesMaskedChannel(const Cluster& cl) const
+{
+  auto found = mMaskedChannels.find(cl.deId);
+  if (found == mMaskedChannels.end()) {
+    return false;
+  }
+
+  double nSigmas = 4.;
+
+  for (auto& area : found->second) {
+    if (std::abs(cl.xCoor - area.getCenterX()) < nSigmas * cl.getEX() + area.getHalfSizeX() &&
+        std::abs(cl.yCoor - area.getCenterY()) < nSigmas * cl.getEY() + area.getHalfSizeY()) {
+      return true;
+    }
+  }
+  return false;
+}
+
 void HitMapBuilder::buildTrackInfo(Track& track, gsl::span<const Cluster> clusters) const
 {
   std::vector<int> firedFEEIdMT11, nonFiredFEEIdMT11;
-  bool outsideAcceptance = false;
+  bool badForEff = false;
   for (int ich = 0; ich < 4; ++ich) {
     auto icl = track.getClusterMatchedUnchecked(ich);
     if (icl >= 0) {
       auto& cl = clusters[icl];
       firedFEEIdMT11.emplace_back(getFEEIdMT11(cl.xCoor, cl.yCoor, cl.deId));
-      // auto localPt = mHitFinder.getGeometryTransformer().globalToLocal(cl.deId, cl.xCoor, cl.yCoor, cl.zCoor);
       for (int icath = 0; icath < 2; ++icath) {
         if (cl.isFired(icath)) {
           track.setFiredChamber(ich, icath);
         }
       }
     } else {
-      auto impactPts = mHitFinder.getLocalPositions(track, ich);
+      auto impactPts = mHitFinder.getLocalPositions(track, ich, true);
       for (auto& impactPt : impactPts) {
         auto feeIdMT11 = getFEEIdMT11(impactPt.xCoor, impactPt.yCoor, impactPt.deId);
         if (feeIdMT11 >= 0) {
           nonFiredFEEIdMT11.emplace_back(feeIdMT11);
+          if (matchesMaskedChannel(impactPt)) {
+            badForEff = true;
+          }
         } else {
-          outsideAcceptance = true;
+          badForEff = true;
         }
       }
     }
   }
   track.setFiredFEEId(firedFEEIdMT11.front());
-  int effFlag = outsideAcceptance ? 0 : getEffFlag(firedFEEIdMT11, nonFiredFEEIdMT11);
+  int effFlag = badForEff ? 0 : getEffFlag(firedFEEIdMT11, nonFiredFEEIdMT11);
   track.setEfficiencyFlag(effFlag);
 }
 

--- a/Detectors/MUON/MID/Tracking/src/HitMapBuilder.cxx
+++ b/Detectors/MUON/MID/Tracking/src/HitMapBuilder.cxx
@@ -46,10 +46,18 @@ bool HitMapBuilder::crossCommonElement(const std::vector<int>& fired, const std:
   return false;
 }
 
-int HitMapBuilder::getEffFlag(const std::vector<int>& firedRPCLines, const std::vector<int>& nonFiredRPCLines, const std::vector<int>& firedLocIds, const std::vector<int>& nonFiredLocIds) const
+int HitMapBuilder::getEffFlag(const std::vector<int>& firedFEEIdMT11, const std::vector<int>& nonFiredFEEIdMT11) const
 {
+  std::vector<int> firedRPCLines, nonFiredRPCLines;
+  for (auto& feeId : firedFEEIdMT11) {
+    firedRPCLines.emplace_back(detparams::getDEIdFromFEEId(feeId));
+  }
+  for (auto& feeId : nonFiredFEEIdMT11) {
+    nonFiredRPCLines.emplace_back(detparams::getDEIdFromFEEId(feeId));
+  }
+
   if (crossCommonElement(firedRPCLines, nonFiredRPCLines)) {
-    if (crossCommonElement(firedLocIds, nonFiredLocIds)) {
+    if (crossCommonElement(firedFEEIdMT11, nonFiredFEEIdMT11)) {
       return 3;
     }
     return 2;
@@ -57,27 +65,25 @@ int HitMapBuilder::getEffFlag(const std::vector<int>& firedRPCLines, const std::
   return 1;
 }
 
-int HitMapBuilder::getLocId(double xp, double yp, uint8_t deId) const
+int HitMapBuilder::getFEEIdMT11(double xp, double yp, uint8_t deId) const
 {
   auto stripIndex = mMapping.stripByPosition(xp, yp, 0, deId, false);
   if (stripIndex.isValid()) {
-    return mMapping.getBoardId(stripIndex.line, stripIndex.column, deId);
+    auto deIdMT11 = detparams::getDEId(detparams::isRightSide(deId), 0, detparams::getRPCLine(deId));
+    return detparams::getUniqueFEEId(deIdMT11, stripIndex.column, stripIndex.line);
   }
-  return 0;
+  return -1;
 }
 
 void HitMapBuilder::buildTrackInfo(Track& track, gsl::span<const Cluster> clusters) const
 {
-  std::vector<int> firedLocIds, nonFiredLocIds;
-  std::vector<int> firedDeIds, firedRPCLines, nonFiredRPCLines;
+  std::vector<int> firedFEEIdMT11, nonFiredFEEIdMT11;
   bool outsideAcceptance = false;
   for (int ich = 0; ich < 4; ++ich) {
     auto icl = track.getClusterMatchedUnchecked(ich);
     if (icl >= 0) {
       auto& cl = clusters[icl];
-      firedDeIds.emplace_back(cl.deId);
-      firedRPCLines.emplace_back(detparams::getRPCLine(cl.deId));
-      firedLocIds.emplace_back(getLocId(cl.xCoor, cl.yCoor, cl.deId));
+      firedFEEIdMT11.emplace_back(getFEEIdMT11(cl.xCoor, cl.yCoor, cl.deId));
       // auto localPt = mHitFinder.getGeometryTransformer().globalToLocal(cl.deId, cl.xCoor, cl.yCoor, cl.zCoor);
       for (int icath = 0; icath < 2; ++icath) {
         if (cl.isFired(icath)) {
@@ -87,20 +93,17 @@ void HitMapBuilder::buildTrackInfo(Track& track, gsl::span<const Cluster> cluste
     } else {
       auto impactPts = mHitFinder.getLocalPositions(track, ich);
       for (auto& impactPt : impactPts) {
-        nonFiredRPCLines.emplace_back(detparams::getRPCLine(impactPt.deId));
-        nonFiredLocIds.emplace_back(getLocId(impactPt.xCoor, impactPt.yCoor, impactPt.deId));
-        if (nonFiredLocIds.back() == 0) {
+        auto feeIdMT11 = getFEEIdMT11(impactPt.xCoor, impactPt.yCoor, impactPt.deId);
+        if (feeIdMT11 >= 0) {
+          nonFiredFEEIdMT11.emplace_back(feeIdMT11);
+        } else {
           outsideAcceptance = true;
         }
       }
     }
   }
-  track.setFiredDeId(firedDeIds.front());
-  track.setFiredLocalBoard(firedLocIds.front());
-  int effFlag = 0;
-  if (!outsideAcceptance) {
-    effFlag = getEffFlag(firedRPCLines, nonFiredRPCLines, firedLocIds, nonFiredLocIds);
-  }
+  track.setFiredFEEId(firedFEEIdMT11.front());
+  int effFlag = outsideAcceptance ? 0 : getEffFlag(firedFEEIdMT11, nonFiredFEEIdMT11);
   track.setEfficiencyFlag(effFlag);
 }
 

--- a/Detectors/MUON/MID/Workflow/include/MIDWorkflow/TrackerSpec.h
+++ b/Detectors/MUON/MID/Workflow/include/MIDWorkflow/TrackerSpec.h
@@ -23,7 +23,7 @@ namespace o2
 {
 namespace mid
 {
-framework::DataProcessorSpec getTrackerSpec(bool isMC);
+framework::DataProcessorSpec getTrackerSpec(bool isMC, bool checkMasked);
 }
 } // namespace o2
 

--- a/Detectors/MUON/MID/Workflow/src/TrackerSpec.cxx
+++ b/Detectors/MUON/MID/Workflow/src/TrackerSpec.cxx
@@ -18,6 +18,7 @@
 
 #include <chrono>
 #include "Framework/DataRefUtils.h"
+#include "Framework/CCDBParamSpec.h"
 #include "Framework/ConfigParamRegistry.h"
 #include "Framework/ControlService.h"
 #include "Framework/Logger.h"
@@ -43,7 +44,7 @@ namespace mid
 class TrackerDeviceDPL
 {
  public:
-  TrackerDeviceDPL(std::shared_ptr<o2::base::GRPGeomRequest> gr, bool isMC) : mGGCCDBRequest(gr), mIsMC(isMC) {}
+  TrackerDeviceDPL(std::shared_ptr<o2::base::GRPGeomRequest> gr, bool isMC, bool checkMasked) : mGGCCDBRequest(gr), mIsMC(isMC), mCheckMasked(checkMasked) {}
   ~TrackerDeviceDPL() = default;
 
   void init(o2::framework::InitContext& ic)
@@ -101,6 +102,12 @@ class TrackerDeviceDPL
 
   void finaliseCCDB(o2::framework::ConcreteDataMatcher& matcher, void* obj)
   {
+    if (mCheckMasked && matcher == of::ConcreteDataMatcher(header::gDataOriginMID, "MASKED_CHANNELS", 0)) {
+      LOG(info) << "Update MID_MASKED_CHANNELS";
+      auto* badChannels = static_cast<std::vector<ColumnData>*>(obj);
+      mHitMapBuilder->setMaskedChannels(*badChannels);
+      return;
+    }
     if (o2::base::GRPGeomHelper::instance().finaliseCCDB(matcher, obj)) {
       return;
     }
@@ -109,6 +116,7 @@ class TrackerDeviceDPL
  private:
   void updateTimeDependentParams(o2::framework::ProcessingContext& pc)
   {
+    // Triggers finalizeCCDB
     o2::base::GRPGeomHelper::instance().checkUpdates(pc);
     static bool initOnceDone = false;
     if (!initOnceDone) {
@@ -120,10 +128,12 @@ class TrackerDeviceDPL
       }
       mHitMapBuilder = std::make_unique<HitMapBuilder>(geoTrans);
     }
+    pc.inputs().get<std::vector<ColumnData>*>("mid_bad_channels_forTracks");
   }
 
   bool mIsMC = false;
   bool mKeepAll = false;
+  bool mCheckMasked = false;
   TrackLabeler mTrackLabeler{};
   std::shared_ptr<o2::base::GRPGeomRequest> mGGCCDBRequest;
   std::unique_ptr<Tracker> mTracker{nullptr};
@@ -134,9 +144,12 @@ class TrackerDeviceDPL
   unsigned int mNROFs{0};                         /// Total number of processed ROFs
 };
 
-framework::DataProcessorSpec getTrackerSpec(bool isMC)
+framework::DataProcessorSpec getTrackerSpec(bool isMC, bool checkMasked)
 {
-  std::vector<of::InputSpec> inputSpecs{of::InputSpec{"mid_clusters", "MID", "CLUSTERS"}, of::InputSpec{"mid_clusters_rof", "MID", "CLUSTERSROF"}};
+  std::vector<of::InputSpec> inputSpecs;
+  inputSpecs.emplace_back("mid_clusters", header::gDataOriginMID, "CLUSTERS");
+  inputSpecs.emplace_back("mid_clusters_rof", header::gDataOriginMID, "CLUSTERSROF");
+  inputSpecs.emplace_back("mid_bad_channels_forTracks", header::gDataOriginMID, "MASKED_CHANNELS", 0, of::Lifetime::Condition, of::ccdbParamSpec("MID/Calib/BadChannels"));
   auto ggRequest = std::make_shared<o2::base::GRPGeomRequest>(false,                             // orbitResetTime
                                                               false,                             // GRPECS=true
                                                               false,                             // GRPLHCIF
@@ -146,23 +159,23 @@ framework::DataProcessorSpec getTrackerSpec(bool isMC)
                                                               inputSpecs,
                                                               true);
   std::vector<of::OutputSpec> outputSpecs{
-    of::OutputSpec{"MID", "TRACKS"},
-    of::OutputSpec{"MID", "TRACKCLUSTERS"},
-    of::OutputSpec{"MID", "TRACKROFS"},
-    of::OutputSpec{"MID", "TRCLUSROFS"}};
+    of::OutputSpec{header::gDataOriginMID, "TRACKS"},
+    of::OutputSpec{header::gDataOriginMID, "TRACKCLUSTERS"},
+    of::OutputSpec{header::gDataOriginMID, "TRACKROFS"},
+    of::OutputSpec{header::gDataOriginMID, "TRCLUSROFS"}};
 
   if (isMC) {
-    inputSpecs.emplace_back(of::InputSpec{"mid_clusterlabels", "MID", "CLUSTERSLABELS"});
+    inputSpecs.emplace_back(of::InputSpec{"mid_clusterlabels", header::gDataOriginMID, "CLUSTERSLABELS"});
 
-    outputSpecs.emplace_back(of::OutputSpec{"MID", "TRACKLABELS"});
-    outputSpecs.emplace_back(of::OutputSpec{"MID", "TRCLUSLABELS"});
+    outputSpecs.emplace_back(of::OutputSpec{header::gDataOriginMID, "TRACKLABELS"});
+    outputSpecs.emplace_back(of::OutputSpec{header::gDataOriginMID, "TRCLUSLABELS"});
   }
 
   return of::DataProcessorSpec{
     "MIDTracker",
     {inputSpecs},
     {outputSpecs},
-    of::adaptFromTask<o2::mid::TrackerDeviceDPL>(ggRequest, isMC),
+    of::adaptFromTask<o2::mid::TrackerDeviceDPL>(ggRequest, isMC, checkMasked),
     of::Options{{"mid-tracker-keep-best", of::VariantType::Bool, false, {"Keep only best track (default is keep all)"}}}};
 }
 } // namespace mid

--- a/Detectors/MUON/MID/Workflow/src/reco-workflow.cxx
+++ b/Detectors/MUON/MID/Workflow/src/reco-workflow.cxx
@@ -84,7 +84,7 @@ WorkflowSpec defineDataProcessing(ConfigContext const& cfgc)
   }
   specs.emplace_back(o2::mid::getClusterizerSpec(!disableMC, dataDesc, rofDesc, labelsDesc));
   if (!disableTracking) {
-    specs.emplace_back(o2::mid::getTrackerSpec(!disableMC));
+    specs.emplace_back(o2::mid::getTrackerSpec(!disableMC, !disableFiltering));
   }
   if (!disableFile) {
     if (disableTracking) {

